Update the readme with a known limitations section (#39)
mendersoftware/mender-python-client: README.md

## Workings

The _client_ in daemon mode does periodic checks for updates at a configurable
interval, and downloads the Artifact for the deployment to a given location on
the device. Then control is passed over to the _sub-updater_ through calling the
script `/usr/share/mender/install <path-to-downloaded-artifact>`.

This forks off a child-process, which then executes the install script. The
_client_ itself, will then exit, since all its obligations are now fulfilled.
If your script has been started, by say _systemd_, make sure that the
sub-process is not killed upon exit. This can be done by adding:

```
...
[service]
...
KillMode=process
```

to your unit file specification. For other init systems, keep this in mind.

Following upon this, it is now the responsibility of the _sub-updater_ to unpack

> the Artifact, and install it to the passive partition, reboot the device, commit

Reviewer: ```suggestion
the Artifact, install it to the passive partition, reboot the device, and commit
```

device_type=<some-device-type>
```

## Known limitations

The _Mender Python Client_ is not a fully fledged _Mender Client_ updater, like
the original [_Mender Client_](https://github.com/mendersoftware/mender). As
such, there are certain limitations to keep in mind, when incorporating it into
your device setup.

For one, the _Mender Python Client_ holds no responsibility over either the
reliability of the downloaded _Mender Artifact_, nor does it concern itself with

> anything which happens after the Artifact is downloaded, and the _install

Reviewer: ```suggestion
anything which happens after the Artifact is downloaded and the _install
```

script_ has been spawned. All of this has to be managed by the sub-updater. This
includes starting the daemon back up after it has exited.

> Secondly, the _Mender Python Client_ currently only reports the following update
> states back to the Mender server:

Reviewer: ```suggestion
Secondly, the _Mender Python Client_ currently only reports the following deployment
status back to the Mender server:
```

To be aligned with the API endpoint name

such, there are certain limitations to keep in mind, when incorporating it into
your device setup.

For one, the _Mender Python Client_ holds no responsibility over either the
reliability of the downloaded _Mender Artifact_, nor does it concern itself with
anything which happens after the Artifact is downloaded, and the _install
script_ has been spawned. All of this has to be managed by the sub-updater. This
includes starting the daemon back up after it has exited.

Secondly, the _Mender Python Client_ currently only reports the following deployment
status back to the Mender server:

* Downloading
* Success
* Failure

### State-machine

The _Mender Python Client_ can run as a daemon, with the _daemon_ command,
however, only a subset of the states are supported, due to the handoff to the
'install' script. The states supported are:

* **Idle**: The Mender client idles and waits for the next action to handle. At this stage, no communication with the server, or downloads are in progress.
* **Sync**: At this stage the Mender client will either send or update its inventory to the server, or check if an update is available. This requires communication with the server.

> * **Download**: When an update is ready at the server side Mender downloads it (streams it) to the inactive rootfs partition.

Reviewer: ```suggestion
* **Download**: When an update is ready at the server side Mender downloads it (streams it) to a designated area in the data partition.
```
It is downloaded to /var/lib, right?

> * **ArtifactInstall**: Swap the active and inactive partitions through interaction with the bootloader.

Reviewer: Call the sub-updater to fulfill the installation?

The remaining functionality has to be supported by the `sub-updater`.

> ![State-machine](docs/mender-python-client-state-machine.pdf)

Reviewer: File missing
